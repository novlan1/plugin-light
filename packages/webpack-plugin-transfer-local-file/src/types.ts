export type ITransferLocalFileOptions= {
  // 是否修改打包产物中的引用位置，默认 false
  isModifyRef?: boolean;
  // 处理的文件夹列表
  adapterDirs?: Array<string>;
};
