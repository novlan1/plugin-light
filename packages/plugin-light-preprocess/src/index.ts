export { preprocessFile, preprocessFileSync, preprocess } from './preprocess';
