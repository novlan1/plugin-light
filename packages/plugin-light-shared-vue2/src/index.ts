export {
  shouldUseLoader,
} from './loader-options';


export {
  parseSFC,
  composeSFC,
} from './sfc';

