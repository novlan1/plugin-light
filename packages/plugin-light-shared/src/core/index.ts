export * from './cross-game-style';
export * from './gen-version';
export * from './ifdef';
export * from './inject-dynamic-style';
export * from './replace-vue-directive';
export * from './v-lazy';
