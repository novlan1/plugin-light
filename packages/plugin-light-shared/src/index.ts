export * from './core';


export {
  checkBundleAnalyze,
  checkDebugMode,
} from './bundle-analyze';

export {
  PLATFORM_MAP,
  ALL_PLATFORM,
  PLATFORMS_ALL,
  PLATFORMS_MP,
  HTML_MAP,
  CSS_MAP,
  CSS_POSTFIX_MAP,
  CDN_MAP,
} from './config';

export {
  DEFAULT_TRANSPILE_DEPENDENCIES,
  DEFAULT_ADAPTER_DIRS,
  AEGIS_EXTERNAL_SCRIPT_LINK,
  UNI_SIMPLE_ROUTER_SCRIPT_LINK,
  EXTERNAL_LINK_MAP,
} from './config-project';

export {
  ROOT_NAME,
  saveJsonToLog,
  createLogDir,
  normalizePath,
  updateAssetSource,
  removeFirstSlash,
  parseSetDeps,
  getRelativePath,
  getUniCliCache,
} from './helper';


export {
  DEFAULT_CONTEXT_OBJECT,
} from './context';


export {
  getDeps,
} from './deps';


export {
  findDependencies,
} from './find-dependencies';


export {
  checkH5,
} from './h5';


export {
  getLoaderFile,
  getLoaderProdFile,
} from './loader-file';

export {
  saveLoaderLog,
  recordLoaderLog,
} from './loader-log';

export {
  getProjectName,
  getSubProjectName,
} from './project-name';

export {
  updateManifest,
  revertManifest,
} from './replace-manifest';

export {
  getRootDir,
} from './root';


export {
  getSubProjectConfig,
  getSubProjectRoot,
} from './sub-project';
