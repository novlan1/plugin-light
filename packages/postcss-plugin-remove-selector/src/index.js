const postcss = require('postcss');

const shouldHandle = (configList = [], current) => configList.find(item => item.file.test(current));

module.exports = postcss.plugin('postcss-plugin-remove-selector', (opts = {}) => {
  const { list } = opts;
  return function (root, result) {
    const fileName = result.opts.from;
    const found = shouldHandle(list, fileName);

    if (!found) {
      return;
    }
    const keySelector = found.excludes?.not || [];
    console.log('[Postcss Plugin Remove Select] handling ', fileName);

    root.walkRules((rule) => {
      if (!keySelector.includes(rule.selector)) {
        rule.remove();
      }

      // rule.walkDecls((decl) => {
      //   console.log(`${decl.prop} = ${decl.value}`);
      // });
    });
  };
});
