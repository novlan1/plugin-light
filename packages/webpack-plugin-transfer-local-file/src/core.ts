const IGNORE_FILE_LIST = [
  'common/vendor.js',
  'common/runtime.js',
  'common/main.js',
];
const isIgnore = (str: string) => IGNORE_FILE_LIST.find(item => str.startsWith(item));


export function replaceSource(source: string, adapterDirs: Array<string>) {
  const replaceReg = new RegExp(`\\.\\./\\.\\./(${adapterDirs.join('|')})(/)(.+?)('|"|/)`, 'g');
  source = source.replace(replaceReg, (...args) => {
    const { 0: origin, 1: localName, 2: slash, 3: innerDir, 4: slashOrQuote } = args;
    const dir = `${localName}${slash}${innerDir}`;
    if (isIgnore(dir)) {
      return origin;
    }
    return `${localName}${slash}${innerDir}${slashOrQuote}`;
  });

  return source;
}
