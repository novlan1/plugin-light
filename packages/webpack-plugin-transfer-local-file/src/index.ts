export { TransferLocalFilePlugin } from './transfer-local-file';
export * from './types';
