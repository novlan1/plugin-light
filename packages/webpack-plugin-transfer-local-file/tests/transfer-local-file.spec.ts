import { replaceSource } from '../src/core';

const baseStr = '(global["webpackJsonp"] =require(\'../../common/abc/123.js\');(global["webpackJsonp"] =';

describe('replaceSource', () => {
  const adapterDirs = [
    'comm',
    'common',
    'component',
    'components',
    'logic',
    'local-logic',
    'local-component',
    'login',
    'pages',
    'static',
  ];

  it('replaceSource', () => {
    const source = `
require('../../common/123.js');(global["webpackJsonp"] =require('../../common/123.js');(global["webpackJsonp"] =
require('../../common/vendor.js');(global["webpackJsonp"] =require('../../common/vendor.js');(global["webpackJsonp"] =
require('../../common/abc/123.js');${baseStr}
require('../../local-logic/abc/123.js');${baseStr}
require('../../components/abc/123.js');${baseStr}
require('../../component/abc/123.js');${baseStr}
require('../../comm/abc/123.js');${baseStr}
require('../../comm/login/123.js');${baseStr}
`;
    expect(replaceSource(source, adapterDirs)).toMatchSnapshot();
  });
});
