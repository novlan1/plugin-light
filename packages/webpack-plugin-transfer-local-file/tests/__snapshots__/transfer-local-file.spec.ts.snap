// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaceSource replaceSource 1`] = `
"
require('common/123.js');(global["webpackJsonp"] =require('common/123.js');(global["webpackJsonp"] =
require('../../common/vendor.js');(global["webpackJsonp"] =require('../../common/vendor.js');(global["webpackJsonp"] =
require('common/abc/123.js');(global["webpackJsonp"] =require('common/abc/123.js');(global["webpackJsonp"] =
require('local-logic/abc/123.js');(global["webpackJsonp"] =require('common/abc/123.js');(global["webpackJsonp"] =
require('components/abc/123.js');(global["webpackJsonp"] =require('common/abc/123.js');(global["webpackJsonp"] =
require('component/abc/123.js');(global["webpackJsonp"] =require('common/abc/123.js');(global["webpackJsonp"] =
require('comm/abc/123.js');(global["webpackJsonp"] =require('common/abc/123.js');(global["webpackJsonp"] =
require('comm/login/123.js');(global["webpackJsonp"] =require('common/abc/123.js');(global["webpackJsonp"] =
"
`;
